velox/common/memory/MappedMemory.cpp

```diff
@@ -231,6 +243,9 @@ bool MappedMemoryImpl::allocate(
         ::free(ptr);
       }
       out.clear();
+      if (userAllocCB != nullptr) {
+        userAllocCB(-bytesToAllocate);
```

Comment:
Does the callback expected to check if input is negative and assume that it means allocation failed? Would it make sense to make this a bit more explicit, by adding a boolean flag to indicate whether call before allocating or on allocation failure? Would be nice to document the semantics as well.